src/praisonai/praisonai/cli/legacy/subcommand_handlers.py

```
@@ -0,0 +1,1312 @@
+    except ImportError as e:
+        print(f"[red]ERROR: Failed to import mcp module: {e}[/red]")
+        print("Make sure praisonaiagents is installed: pip install praisonaiagents")
+    except Exception as e:
+        print(f"[red]ERROR: MCP command failed: {e}[/red]")
+
+# Compiled regex patterns for sensitive data detection (compiled once, zero runtime cost)
+_SENSITIVE_PATTERNS = None
+
+def _get_sensitive_patterns():
+    """Lazy-load and compile sensitive patterns only when needed."""
+    if _SENSITIVE_PATTERNS is None:
```

Ruff (0.15.20)

[error] Local variable `_SENSITIVE_PATTERNS` referenced before assignment

(F823)